### DISCUSSION TITLE
Security audit?

### DISCUSSION THREAD

**Discussion by github2099** [rustdesk/rustdesk#4968] [General]
any being planned?

**Comment by dinger1986** [comment]
not at the moment, rust has audits built in, there is a plan to implement these automatically on pulls but a full external audit is alot further down on the roadmap, its opensource and people can audit it themselves, or can fund/help fund it.

**Comment by kassemchakaroun** [comment]
Any news about Security audit ?

**Reply by Rusty-Weasel** [reply]
nope, but i just can give u one,
git clone -> open project in ide and  run -> **cargo audit** -> and.... you got this result:

![image](https://github.com/rustdesk/rustdesk/assets/61587283/23348df0-d84d-4d4a-be23-ab9deebbe8ad)

top = oss version client
bottom = oss version server

**Reply by Rusty-Weasel** [reply]
it's just normal rust stuff, nothing special.

build up your dev environment, in my case i like to use jetbrains rustrover, but u also can do it with vscode, or just vim.
after you have rust dev environment, you just need https://crates.io/crates/cargo-audit , rly ez to install,
after that go into your project root and just execute "**cargo audit**", it will check all your deps on one of the already known (about 600 at moment i think) issues by RustSec. 

You then also get links, to each vul, with detailed information about and if there is already a solution to fix it.

Things like that are really absolute beginners level and was one of the main things i learned as first on nearly every rust course.
There are further tools which can check, for licencse problems or to many **unsafe** or **unwrap** uses in your dep an code, and many more, list ist daily growing.
